### PR TITLE
docs: update contributing guidelines

### LINKED ISSUE #117: Create GitHub team with project maintainers?
I've created a GitHub team [terraform-baseline-maintainers](https://github.com/orgs/equinor/teams/terraform-baseline-maintainers).

This team should be configured as code owners in all Baseline repos, so that all changes will go through them.

This means that anyone in the terraform-baseline team will still be able to contribute directly to all module repositories (without creating a fork), however contributions must be approved by one of the maintainers.

## Tasks

- [x] Update contributing guidelines to clarify how to become part of maintainer team
- [x] Update code owners for all module repositories

### PR DESCRIPTION
**Pull request by hknutsen**
- Clarify roles and responsibilities
- Update code owners

Closes #117

### CODE REVIEW

**Comment by hknutsen** [comment]
Suggestions:
- Contributors can add new modules.
- Maintainers still approve PRs in new modules.
- Admins publish modules to registry.

**Comment by hknutsen** [comment]
Who creates new workflows in GitHub project?

**Review by chbul — DISMISSED** [review]

**Review by sebastianlolv — DISMISSED** [review]
LGTM

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,13 +1,63 @@
...
+
+An administrator must:
+
+- Be a member of the [@equinor/terraform-baseline-admins](https://github.com/orgs/equinor/teams/terraform-baseline-admins) team.
+
+An administrator should:
+
+- Configure module repositories.
+- Public modules to Terraform Registry.
+- Recruit people to the above roles 🤗
```
Comment (helenakallekleiv) [inline CONTRIBUTING.md:63]: Should we include that either Adminitrators or Administrators+Maintainers should also have a role for when to archive modules?

Comment (hknutsen) [reply CONTRIBUTING.md]: According to [this overview](https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/managing-repository-roles/repository-roles-for-an-organization#permissions-for-each-role), only admins have the required permissions to archive repositories, so I'll add this to the responsibilities of `Administrators`.

**Review by sebastianlolv — APPROVED** [review]

**Review by helenakallekleiv — APPROVED** [review]

**Pull request merged**